src/rgw/rgw_rados.cc

```diff
@@ -11296,18 +11296,21 @@ int RGWRados::omap_get_all(rgw_obj& obj, bufferlist& header,
     return r;
   }
 
-  const int count = 1024;  // ?
+#define MAX_OMAP_GET_ENTRIES 1024
+  const int count = MAX_OMAP_GET_ENTRIES;
   string start_after;
+  bool done = false;
 
-  while (true) {
+  while (!done) {
     std::map<string, bufferlist> t;
-    r = ref.ioctx.omap_get_vals(ref.oid, start_after, count, &m);
+    r = ref.ioctx.omap_get_vals(ref.oid, start_after, count, &t);
     if (r < 0) {
       return r;
     }
     if (t.empty()) {
       break;
     }
+    done = (t.size() < count);
```

Comment:
(For example, there is a now an omap get limit that's enforced at the osd to work around client code like this)